starport/pkg/xfilepath/xfilepath.go

```diff
@@ -0,0 +1,34 @@
+package xfilepath
+
+import (
+	"os"
+	"path/filepath"
+)
+
+type PathRetriever func() (path string, err error)
+
+func Path(path string) PathRetriever {
+	return func() (string, error) { return path, nil }
+}
+
+func PathWithError(path string, err error) PathRetriever {
+	return func() (string, error) { return path, err }
+}
+
+func Join(paths ...PathRetriever) PathRetriever {
+	return func() (string, error) {
+		var components []string
```

Comment:
A small optimization can be done here.

We can move the content of this code block (line 20-28) to the upper level. Cache the path and error there to two different vars and return them in this func as `return func() (string, error) { return path, err }`.